how-tos/maintain/upgrade-notes-sql-mongodb/upgrade-to-v4.md

```diff
@@ -37,6 +37,10 @@ npm install forest-express-mongoose@4.1.2
 In case of a regression introduced in Production after the upgrade, a rollback to your previous agent version 3 is the fastest way to restore your admin panel.
 {% endhint %}
 
+{% hint style="info" %}
+You must upgrade your agent version in development, then you should commit the code changes (packages.json, source code, .forestadmin-schema.json etc.) to push it on other environments (Production, Staging, Test,...). Pull the code in each server, install, build and restart server
+{% endhint %}
```

Comment:
A link to the appropriate documentation about the `.forestadmin-schema.json` file management would highlight the point.

Link examples:
- https://docs.forestadmin.com/documentation/extra-help/maintain/upgrade-notes-sql-mongodb/upgrade-to-v3#schema-versioning
- https://docs.forestadmin.com/documentation/reference-guide/models#the-.forestadmin-schema.json-file

Reply:
The sentence should be simplified.
We basically teaching developers how to deploy, whereas we just want to let them know that the schema file needs to reflect the upgrade (via development upgrade, and then versioning).

Reply:
This comment is also relevant for other stacks.